# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerThread.h
# Commit: 36ef70762b74
# Full Hash: 36ef70762b74b3c6b8bd0f26c57ab4b54467f64b
# Author: Bill McCloskey <billm@mozilla.com>
# Date: 2017-08-17 23:17:14
# Regressor Bug: 1382922
# File Overlap Count: 3
# Description:
#   Bug 1382922 - Refactor event queue to allow multiple implementations (r=erahm)
#   
#   This patch refactors the nsThread event queue to clean it up and to make it easier to restructure. The fundamental concepts are as follows:
#   
#   Each nsThread will have a pointer to a refcounted SynchronizedEventQueue. A SynchronizedEQ takes care of doing the locking and condition variable work when posting and popping events. For the actual storage of events, it delegates to an AbstractEventQueue data structure. It keeps a UniquePtr to the AbstractEventQueue that it uses for storage.
# ==============================================================================

diff -r ca6618d0bc17 -r 36ef70762b74 dom/workers/WorkerThread.h
--- a/dom/workers/WorkerThread.h	Tue Aug 01 12:55:21 2017 -0400
+++ b/dom/workers/WorkerThread.h	Tue Jun 20 19:42:13 2017 -0700
@@ -78,14 +78,6 @@
   uint32_t
   RecursionDepth(const WorkerThreadFriendKey& aKey) const;
 
-  // Required for MinGW build #1336527 to handle compiler bug:
-  // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79582
-  NS_IMETHOD
-  RegisterIdlePeriod(already_AddRefed<nsIIdlePeriod> aIdlePeriod) override
-  {
-    return nsThread::RegisterIdlePeriod(already_AddRefed<nsIIdlePeriod>(aIdlePeriod.take()));
-  }
-
   NS_DECL_ISUPPORTS_INHERITED
 
 private: